shotover-proxy/src/runner.rs

```diff
@@ -77,7 +77,11 @@ impl Runner {
     pub fn with_observability_interface(self) -> Result<Self> {
         let recorder = PrometheusBuilder::new().build_recorder();
         let handle = recorder.handle();
-        metrics::set_boxed_recorder(Box::new(recorder))?;
+
+        // Result::Err can only occur when set_boxed_recorder has already been called.
+        // So we just ignore the error with .ok() as integration tests will cause shotover
+        // to be initialized multiple times thus recalling this function.
+        metrics::set_boxed_recorder(Box::new(recorder)).ok();
```

Comment:
Ah...
Probably this doesnt actually work because some of the state here gets dropped and deinitializes the recorder when shotover shutsdown

Reply:
Oops, I thought the test suite had passed when I approved.

Reply:
I pushed a fix.
The only reason im considering this fix is because the issue existed before, the only difference then was the clear_recorder function wasnt marked as unsafe. refer to https://github.com/metrics-rs/metrics/pull/302
So this is actually an improvement as the unsafe call will now show up in an audit whereas before the issue would have gone unnoticed.

And I really have no idea how to go about fixing it properly.
The proper fix would probably be actually spawning shotover in a separate process, that would fix this issue and a very similar issue in tracing init.
But that has considerable downsides and implementation time.